code/ZAS/Fire.dm

```diff
@@ -34,11 +34,24 @@ Attach to transfer valve and open. BOOM.
 		//testing("[src] ashifying (BFF)!")
 		ashify()
 
+/atom/proc/burnItselfUp()
+	while(on_fire)
```

Comment:
There has got to be a better way than having each object make its own while loop when on fire

Fire sub perhaps?

Reply:
I already have a more comprehensive fix for this on my firecode branch, so this would be temporary anyway
Assuming I ever finish that